ui-admin/src/search/SearchQueryBuilder.tsx

```diff
@@ -35,6 +52,121 @@ export const SearchQueryBuilder = ({
   studyEnvContext: StudyEnvContextT,
   onSearchExpressionChange: (searchExpression: string) => void,
   searchExpression: string
+}) => {
+  const [advancedMode, setAdvancedMode] = useState(false)
+
+  const parseSearchExpError = useMemo(() => {
+    try {
+      if (isEmpty(searchExpression)) {
+        return undefined
+      }
+      parseExpression(searchExpression)
+    } catch (e) {
+      return e as Error
+    }
+  }, [searchExpression])
+
+
+  const [searchResults, setSearchResults] = useState<EnrolleeSearchExpressionResult[]>([])
+
+  const {
+    isLoading: isLoadingSearchResults
+  } = useLoadingEffect(async () => {
+    try {
+      const newResults = await Api.executeSearchExpression(
+        studyEnvContext.portal.shortcode,
+        studyEnvContext.study.shortcode,
+        studyEnvContext.currentEnv.environmentName,
+        searchExpression)
+
+      setSearchResults(newResults)
+    } catch (_) {
+      setSearchResults([])
+    }
+  }, [searchExpression])
+
+  return <div>
+    <div className="mb-2 d-flex flex-row align-items-center justify-content-between">
+      <div>
+        Create a <ZendeskLink doc={DocsKey.SEARCH_EXPRESSIONS}>
+          <FontAwesomeIcon icon={faInfoCircle} className={'me-1'}/>
+        search expression
+        </ZendeskLink> that filters to enrollees that meet a specific criteria.
+      </div>
+      <button
+        className="btn btn-link"
+        onClick={() => setAdvancedMode(!advancedMode)}>
+        {advancedMode ? '(switch to basic view)' : '(switch to advanced view)'}
+      </button>
+    </div>
+    {parseSearchExpError && <div className="alert alert-danger mb-2">
+      {parseSearchExpError.message}
+    </div>}
+    <div className="mb-2">
+      {
+        advancedMode
+          ? <AdvancedQueryBuilder
+            onSearchExpressionChange={onSearchExpressionChange}
+            searchExpression={searchExpression}/>
+          : <BasicQueryBuilder
+            studyEnvContext={studyEnvContext}
+            onSearchExpressionChange={onSearchExpressionChange}
+            searchExpression={searchExpression}
+          />
+      }
+    </div>
+
+    <LoadingSpinner isLoading={isLoadingSearchResults}>
+      <span>
+        {searchResults.length} matching enrollee{searchResults.length !== 1 ? 's' : ''}
```

Comment:
nit: check out the `pluralize` function, it can handle modifying the string plurality for you

Reply:
sweet!